dockstore-webservice/src/main/java/io/dockstore/webservice/languages/WDLHandler.java

```diff
@@ -130,6 +133,14 @@ public Version parseWorkflowContent(String filepath, String content, Set<SourceF
             tempMainDescriptor = File.createTempFile("main", "descriptor", Files.createTempDir());
             Files.asCharSink(tempMainDescriptor, StandardCharsets.UTF_8).write(content);
             try {
+                // Set the language version for the main descriptor source file
+                if (mainDescriptorSourceFile != null) {
+                    final String languageVersion = getLanguageVersion(mainDescriptorSourceFile.getContent());
+                    mainDescriptorSourceFile.setTypeVersion(languageVersion);
+                    // All WDL files used by a workflow must have the same version
+                    version.setDescriptorTypeVersion(languageVersion);
```

Comment:
If I understand correctly, this is an assumption. In the case that the workflow is wrong and contains mixed versions, we would show the version as being the version in the primary. A validation error would only show up later if we implement better validation or ingest validation as a metric? 

(Not a problem, just confirming my understanding)